tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,133 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/CoefficientTransforms.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Gsl.hpp"
+
+// These tests generate a function `u_expected` from a linear superposition of
+// the basis functions, which are then transformed to spectral space
+// (u_coeff) and the coefficients compared with the coefficients set in physical
+// space. The functions are then transformed back (u_back) and compared to the
+// original function, `u_expected`.
+namespace {
+template <Spectral::Basis Basis, Spectral::Quadrature Quadrature, size_t Dim>
+void check_transforms(
+    const Mesh<Dim>& mesh,
+    const std::vector<std::array<size_t, Dim>>& coeffs_to_include) noexcept {
+  CAPTURE(Basis);
+  CAPTURE(Quadrature);
+  CAPTURE(mesh);
+  CAPTURE(coeffs_to_include);
+
+  const auto logical_coords = logical_coordinates(mesh);
```

Comment:
Please add a comment above this line like „Construct a function f(x) = sum_k f_k phi_k(x) in 1D, where the modal coefficients f_k are 1 for the indices k in coeffs_to_include and 0 otherwise. In higher dimensions, construct the Cartesian product of the 1D functions.“ (if this is indeed what is happening)

Reply:
I hope this description is indeed correct?

Reply:
Yes, though I'll explicitly write the 2D and 3D versions.